### DISCUSSION TITLE
Fullstack - Api na rede não gera token

### DISCUSSION THREAD

**Discussion by FBODeveloper** [balta-io/forum#676] [Tira Dúvidas] [answered]
Coloquei a api do curso pra rodar no meu servidor.... tá rodando, gera documentação e tudo mais....
O register salva no banco e tudo
Já consegui gerar o Cookie, porém, nenhum rota, exceto as rotas nativas do Identity, funcionam, conforme erro na mensagem abaixo todas as rotas dá erro 404 not found.

o que será que fiz de errado?

![image](https://github.com/user-attachments/assets/628aeae3-59a6-46bb-9ae3-c0f6e1f3af03)

**Comment by FBODeveloper** [comment, marked as answer]
Consegui gerar o Cookie, porém, qualquer endpoint que não seja o nativo do AspNetCore.Identity dá erro 404 not found.... mesmo os endpoints customizados de logout por exemplo.... tudo dá 404....
Alguém tem regras e configurações que precisa ser feito para a api roda em rede por exemplo?

@andrebaltieri ajuda nóis kkkk

**Comment by FBODeveloper** [comment]
No navegador (no console) dá o erro abaixo:
Erro 404 (Not Found): 
http://192.168.1.249:5000/Account/Login?ReturnUrl=%2Fv1%2Faccount%2Fperfil 404 (Not Found)

essa rota /Account/Login é algo interno ?! pois a rota de login deveria ser /v1/identity/login

o que parece é que ele não resolve essa rota para enviar o cookie(autenticação) para as demais rotas

**Reply by andrebaltieri** [reply]
Bom dia, @FBODeveloper, parece que está em um cenário diferente do curso.... consegue me situar de como estão publicadas suas aplicações?

**Reply by FBODeveloper** [reply]
Basicamente é o Curso...
Localhost tudo certo...funciona tudo perfeitamente...
No server (interno tá... rede aqui na empresa) simplesmente criei um site no ISS (server 2016)... e apontei esse site para a api

na api lá no server eu rodo o dotnet publish --configuration Release
Pego a pasta publish criada e copio os arquivos para a pasta que está apontado o site no ISS

com isso a api roda....
dentro do server em localhost também funciona certinho
só não funciona em rede... (os endpoints criados por mim... como categoria, transacao e empresa que criei para separar os dados)
os endpoints do identity ACREDITO estarem funcionando...pois tanto o register quanto o login não dão erro... sendo que o register cria o usuário e o login retorna 200 e gera o Cookie (no inicio desse issue nem o cookie gerava)....

Estou pesquisando bastante...deve ter algo a ver com https e certificado SSL que não coloquei lá 

Inicialmente não conseguia acessar a API na rede.... ai pesquisando cheguei nesse código que coloquei no BuilderExtension junto ao método AddCrossOrigin

builder.Services.AddCors(options =>
{
    options.AddPolicy(ApiConfiguration.CorsPolicyName,
        builder =>
        {
            builder.SetIsOriginAllowed(origin =>
            {
                var host = new Uri(origin).Host;
                return host.StartsWith("192.168.1.");
            })
            .AllowAnyHeader()
            .AllowAnyMethod()
            .AllowCredentials();
        });
});

**Reply by andrebaltieri** [reply]
Na verdade, não, por que o compartilhamento de cookies não é permitido... logo, um cookie criado no endereço 192.168.20.20 não serve para 192.168.20.21.

Você pode criar um domínio e hostear as aplicações nele, mantê-las no mesmo IP, ou usar uma política de compartilhamento de cookie.

**Reply by FBODeveloper** [reply]
HUmm entendi.... esclarecendo então, esse tipo de autenticação (sem as formas que vc mencionou) devem estar rodando no mesmo server correto?
